### ISSUE TITLE
Can't create non-sensitive vercel_project_environment_variable

### ISSUE THREAD

**Issue by movinfinex** [vercel/terraform-provider-vercel#183]
Hi, I'm using version 1.10.0 of your provider and v1.6.2 of OpenTofu.

With a config like this:

```
resource "vercel_project_environment_variable" "test" {
  project_id = vercel_project.test.id
  target     = ["preview"]
  key        = "TEST"
  value      = "value"
  sensitive  = false
}
```

I get this on apply:

```
│ Error: Provider produced inconsistent result after apply
│
│ When applying changes to vercel_project_environment_variable.test, provider
│ "provider[\"registry.opentofu.org/vercel/vercel\"]" produced an unexpected new value: .sensitive: was cty.False, but now
│ cty.True.
│
│ This is a bug in the provider, which should be reported in the provider's own issue tracker.
```

It works if I remove `sensitive = false`, but then the values can't be seen in the dashboard, of course.

Are you able to reproduce this, or do you need more info from me?

**Comment by dglsparsons** [comment]
Hi @movinfinex, does this work if you omit the `sensitive = false` entirely? I believe that should be creating non-sensitive env vars. I'll look into getting this reproduced ASAP though.

**Comment by dglsparsons** [comment]
Okay, @movinfinex. I can't reproduce this, even on the same opentofu and provider versions. Looking at the provider code, I also don't understand how this could be possible. 

This was my attempt at a reproduction: 
```hcl 
resource "vercel_project" "test" {
  name = "test-acc-project-foo"
}

resource "vercel_project_environment_variable" "test" {
  project_id = vercel_project.test.id
  target     = ["preview"]
  key        = "TEST"
  value      = "value"
  sensitive  = false
}
```

Are you able to provide a repro, or anything else that could be useful?

**Comment by dglsparsons** [comment]
I don't understand this comment either: 
> It works if I remove sensitive = false, but then the values can't be seen in the dashboard, of course.

Without specifying `sensitive`, then the environment variable should default to being `encrypted` only. meaning it can be seen in the UI.

**Comment by movinfinex** [comment]
Okay, I found the issue.

My Vercel team had this set: https://vercel.com/docs/projects/environment-variables/sensitive-environment-variables#environment-variables-policy

Maybe there's a way for the provider to give a helpful error in this case.  I've created a quick docs PR that you can either merge or close, whatever works for you.

> I don't understand this comment either:
By "works" I just meant that the provider could create the variable.  Not being able to read the value was expected.

Thanks for looking into it.

**Comment by dglsparsons** [comment]
Ah, I see. I didn't appreciate that that will affect things. 

It's still very much a provider bug. Although thanks for raising your PR. I'll make some tweaks next week and see if I can get some sensible error messages for this instead!

**Comment by dglsparsons** [comment]
I have a fix currently releasing as v1.10.1. This should log out a friendly message indicating why this is happening. 

It seems the TF provider didn't play nicely with the sensitive environment variable policy at all.
